AoT compilation updates

arknotts opened the pull request:
Changes needed for AoT compilation. Mostly revolving around using ngc compiler rather than tsc. Also added a module at index.ts or the compiler would complain. This is my first time converting something to use AoT compilation but I followed this guide: https://medium.com/@isaacplmann/getting-your-angular-2-library-ready-for-aot-90d1347bcad#.g1570hrdz

Note: I changed the npm build script from `npm run tsc` to `npm run build` since it's no longer using tsc to compile. If you'd rather use something else that's closer to your conventions I'm fine with that, I'll just push another commit.

arknotts commented:
Looks like I need to modify the travis config as well, as it tries to run `npm run tsc`. I'll push another commit tonight.

juergen-vogel commented:
Let me know if you need support. There's a second error on the `transform` function: 

> Not all code paths return a value.

My AoT compiler complains about that.

arknotts commented:
@juergen-vogel Thanks for the offer I'll let you know if I do. That's odd about the `transform` function, mine compiles fine. What version of `@angular/compiler` did it pull down? Either way I see how the transform function should be modified to avoid that error and can include it in the commit.

arknotts commented:
@juergen-vogel I changed the last else statement that should fix your compile issue. I tried it with the `--noImplicitReturns` flag on my machine and it works.

@AndrewPoyntz I updated the travis config file but it still doesn't build on the CI server. Is there a configuration that needs updated to do this? I don't mind changing it back to `npm run tsc` but I feel it's more descriptive of it's purpose now.

juergen-vogel commented:
@arknotts Thanks for your effort! Works fine! 👍
